cpp/src/reductions/minmax.cu

```diff
@@ -0,0 +1,183 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+// The translation unit for reduction `minmax`
+
+#include <thrust/transform_reduce.h>
+#include <cudf/column/column_device_view.cuh>
+#include <cudf/column/column_view.hpp>
+#include <cudf/detail/utilities/device_operators.cuh>
+#include <cudf/reduction.hpp>
+#include <cudf/scalar/scalar_factories.hpp>
+#include <type_traits>
+
+namespace cudf {
+namespace detail {
+
+namespace {
+/**
+ * @brief stores the minimum and maximum
+ * values that have been encountered so far
+ *
+ */
+template <typename T>
+struct minmax_pair {
+  T min_val;
+  T max_val;
+  bool min_valid;
+  bool max_valid;
```

Comment:
Can you use `scalar_device_view` instead ?

Reply:
I'm not sure how this would work. Can you help me out some here? My understanding is that a `scalar_device_view` would need some backing scalar and we don't have that here. Are you suggesting that this turn into a scalar instead of a T?

Reply:
You make good points I didn't consider. I think it's safe to ignore this suggestion.